### PR TITLE
Improve Preferences UI code

### PR DESCRIPTION
**Pull request by SuperDragonXD**
## Description

This PR aims to ease maintainability of the Preferences UI area of Lawnchair.

List of things I (or someone else can) need to do:
- [x] Remove deprecated insets UI
- [x] Modernize Search Bar look
- [x] Promote Font settings to General settings
- [x] Cleanup strings
- [X] Revert two pane layout
- [X] Migrate most components to Material3
- [x] Fix bottom sheet theme

## Type of change

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:x: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by SuperDragonXD** [comment]
I'll slowly update this PR in the following days, although feel free to review on certain parts of this WIP PR.

**Comment by Goooler** [comment]
I suggest we can fully migrate to M3 in the PR.

**Comment by SuperDragonXD** [comment]
Sure.

Also, I'll probably move the 2-pane layout improvements to a different PR since it will require more refactors (probably a complete overhaul) for this PR. Will revert the commits soon-ish

**Comment by SuperDragonXD** [comment]
The PR should be good to go now. I didn't add the previews/docs since we will have a new Preference UI anyway.

File: lawnchair/res/values/strings.xml [hunk 1]
```diff
@@ -18,9 +18,9 @@
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     <string name="iconPackPackageDefault" translatable="false">""</string>
     <!-- Todo: find common strings and centralize them here. Should we make all strings specific though?  -->
-
+    T
```
Comment (Goooler) [inline lawnchair/res/values/strings.xml]:
```suggestion

```

**Review by Goooler — APPROVED** [review]

**Review by MrSluffy — APPROVED** [review]

File: lawnchair/src/app/lawnchair/ui/placeholder/Placeholder.kt [hunk 2]
```diff
@@ -100,10 +103,13 @@ object PlaceholderDefaults {
  * @param contentFadeTransitionSpec The transition spec to use when fading the content
  * on/off screen. The boolean parameter defined for the transition is [visible].
  */
+// TODO: need to migrate to Modifier.Node, see https://mrmans0n.github.io/compose-rules/rules/#avoid-modifier-extension-factory-functions
+@Suppress("ktlint:compose:modifier-composable-check")
```
Comment (Goooler) [inline lawnchair/src/app/lawnchair/ui/placeholder/Placeholder.kt:107]: Only one more thing needs to be migrated.

**Review by Goooler — APPROVED** [review]

**Pull request merged**